lib/src/main/java/graphql/nadel/NextgenEngine.kt

```diff
@@ -383,7 +390,23 @@ internal class NextgenEngine(
 
         if (serviceExecResult is NadelIncrementalServiceExecutionResult) {
```

Comment:
I think you could move all that to `executeTopLevelField` to avoid having to pass the extra params down, and also to keep both calls to `transform` (for the initial and incremental data) close together